File: audit/tasks/contract-authoring/abs_contract/controls/abs_contract.go
//go:build ignore

// Go has no contract construct — this control implements the function only;
// the contract is the Aril-specific delta the task measures (methodology §5).
package main

import "fmt"

func abs(n int) int {
	if n < 0 {
		return -n
	}
	return n
}

func main() {
	fmt.Println(abs(-5))
	fmt.Println(abs(3))
	fmt.Println(abs(0))
}

File: audit/tasks/contract-authoring/abs_contract/controls/abs_contract.ts
// Reference-language control (TypeScript). Statement only — not machine-validated
// in this environment (no Node/tsc). Same I/O contract as the task.
// TypeScript has no contract construct — function only; the contract is the
// Aril-specific delta the task measures (methodology §5).
function abs(n: number): number {
  return n < 0 ? -n : n;
}
console.log(abs(-5));
console.log(abs(3));
console.log(abs(0));
